packages/openapi/src/decorators.ts

```diff
@@ -139,8 +150,8 @@ export function setExtension(program: Program, entity: Type, extensionName: stri
  * @param program Program
  * @param entity Type
  */
-export function getExtensions(program: Program, entity: Type): ReadonlyMap<ExtensionKey, any> {
-  return program.stateMap(openApiExtensionKey).get(entity) ?? new Map<ExtensionKey, any>();
+export function getExtensions(program: Program, entity: Type): Array<ExtensionRecord> {
```

Comment:
I think changing that will be quite a breaking change for anyone using the API.

was there an issue with saving the new data into `program.stateMap(openApiDeepExtensionKey);` or something like that and exposing a separate function that return those.